frontend/src/metabase/visualizations/components/settings/ChartSettingHideButton.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { t } from "ttag";
+import cloneDeep from "lodash.clonedeep";
+import _ from "underscore";
+
+import Button from "metabase/core/components/Button";
+import { updateCardVisualizationSettings } from "metabase/query_builder/actions";
+import type { Column } from "metabase-types/types/Dataset";
+import type { State } from "metabase-types/store";
+
+import { TableColumnOrderSetting } from "metabase-types/api";
+import { normalizeFieldRef } from "metabase-lib/queries/utils/dataset";
+
+interface ChartSettingHideButtonProps {
+  column: Column;
+  columns: Array<Column>;
+  onClose?: () => void;
+}
+
+export function ChartSettingHideButton({
+  column,
+  columns,
+  onClose,
```

Comment:
I'm worried that it directly binds our Table visualization to the query builder at least through the selector `state.qb.card?.visualization_settings`, but the visualization ideally should be ready to use in other places

Reply:
I'm not as familiar with modifying settings directly from the table component, but it shouldn't really be any different than how we change any of the other column settings from the header. I'll pull down the branch and play with it a bit

Reply:
Alright, I think I'm up to speed, and yes, we can do what @alxnddr has above. That 4th argument in `getProps` is what we call `extra`, and it can change depending on the situation. But in the case of nested settings (column and series settings), it contains the overall series and settings objects. In other situation it can contain things like whether we're rendering things in a dashboard vs. the query builder, or the raw series if we're working with a transformed series.

Reply:
I guess the other side of it is how you write the new data. In theory, you can use the onChangeSettings prop that is passed to Chart Settings widgets to do something like: `onChangeSettings('table.columns": [...new array])`

Reply:
I don't love it, but I'm not sure how else you do it :/

Reply:
`onChange` updates your current setting, but `onChangeSetting` allows you to update any setting.

https://github.com/metabase/metabase/blob/716aed38dd365c929340ab94dff50f3849c5f54e/frontend/src/metabase/visualizations/lib/settings.js#L171

Reply:
Regarding the data flow, can confirm that @alxnddr's suggestion of using the `settings` from the `extra` argument to `getProps` works for this use case.

Regarding updating the setting, I tried using `onChangeSettings` and unfortunately it does not work here. It is like @alxnddr said, it updates the nested object in `column_settings`.  I logged input it gave to the underlying `handleChangeSettings` method it calls in `FormatPopover` and this was the result:

<img width="799" alt="Screenshot 2023-03-14 at 8 50 10 AM" src="https://user-images.githubusercontent.com/37751258/225061537-89517983-6e5c-4fac-ba57-fd5dcfd6f3bd.png">